apps/heft/src/utilities/ConfigFile.ts

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { ConfigurationFile, InheritanceType, PathResolutionMethod } from '@rushstack/heft-config-file';
+
+import { ICopyStaticAssetsConfigurationJson } from '../plugins/CopyStaticAssetsPlugin';
+import { IApiExtractorPluginConfiguration } from '../plugins/ApiExtractorPlugin/ApiExtractorPlugin';
+import { ITypeScriptConfigurationJson } from '../plugins/TypeScriptPlugin/TypeScriptPlugin';
+import { HeftConfiguration } from '../configuration/HeftConfiguration';
+import { FileSystem } from '@rushstack/node-core-library';
+
+export enum HeftEvent {
+  clean = 'clean',
+  preCompile = 'pre-compile',
+  compile = 'compile',
+  bundle = 'bundle',
+  postBuild = 'post-build'
+}
+
+export interface IHeftConfigurationJsonEventActionBase {
+  actionKind: string;
+  heftEvent: 'clean' | 'pre-compile' | 'compile' | 'bundle' | 'post-build';
+  actionId: string;
+}
+
+export interface IHeftConfigurationDeleteGlobsEventAction extends IHeftConfigurationJsonEventActionBase {
+  actionKind: 'deleteGlobs';
+  globsToDelete: string[];
+}
+
+export interface IHeftConfigurationJsonPluginSpecifier {
+  plugin: string;
+  options?: object;
+}
+
+export interface IHeftConfigurationJson {
+  eventActions?: IHeftConfigurationJsonEventActionBase[];
+  heftPlugins?: IHeftConfigurationJsonPluginSpecifier[];
+}
+
+export interface IHeftEventActions {
+  deleteGlobs: Map<HeftEvent, IHeftConfigurationDeleteGlobsEventAction[]>;
+}
+
+export class ConfigFile {
+  private static _heftConfigFileLoader: ConfigurationFile<IHeftConfigurationJson> | undefined;
+
+  private static _heftConfigFileEventActionsCache: Map<HeftConfiguration, IHeftEventActions> = new Map<
+    HeftConfiguration,
+    IHeftEventActions
+  >();
+
+  private static _copyStaticAssetsConfigurationLoader:
+    | ConfigurationFile<ICopyStaticAssetsConfigurationJson>
+    | undefined;
+  private static _apiExtractorTaskConfigurationLoader:
+    | ConfigurationFile<IApiExtractorPluginConfiguration>
+    | undefined;
+  private static _typeScriptConfigurationFileLoader:
+    | ConfigurationFile<ITypeScriptConfigurationJson>
+    | undefined;
+
+  public static get heftConfigFileLoader(): ConfigurationFile<IHeftConfigurationJson> {
+    if (!ConfigFile._heftConfigFileLoader) {
+      const schemaPath: string = path.join(__dirname, '..', 'schemas', 'heft.schema.json');
```

Comment:
This class should be called something like `HeftConfigFile`. Otherwise it sounds like an abstract `*.json` loader. It was not obvious that it's the specialized loader for `heft.json`.